src/payment/checkout/payment-form/StripePaymentForm.test.jsx

```diff
@@ -0,0 +1,283 @@
+            <Elements stripe={mockStripe}>
+              <StripePaymentForm
+                handleSubmit={() => {}}
+                onSubmitPayment={() => {}}
+                onSubmitButtonClick={() => {}}
+              />
+            </Elements>
+          </reactRedux.Provider>
+        </AppContext.Provider>
+      </IntlProvider>
+    ));
+    paymentElement = wrapper.find(PaymentElement);
+    submitButton = wrapper.find(PlaceOrderButton);
+  });
+
+  afterEach(() => {
+    jest.restoreAllMocks();
+  });
+
+  describe('load stripe', () => {
+    it('creates the payment element', () => {
+      expect(mockElements.create).toHaveBeenCalledWith('payment', null);
+      expect(paymentElement.length > 0).toBe(true);
+    });
+  });
+
+  describe('getRequiredFields', () => {
+    it('returns expected required fields', () => {
+      const testFormValues = [
+        {
+          firstName: '',
+          lastName: '',
+          address: '',
+          city: '',
+          country: 'UK',
+          cardExpirationMonth: '',
+          cardExpirationYear: '',
+          optionalField: '',
+        },
+        {
+          firstName: '',
+          lastName: '',
+          address: '',
+          city: '',
+          country: 'CA',
+          cardExpirationMonth: '',
+          cardExpirationYear: '',
+          optionalField: '',
+        },
+        {
+          firstName: '',
+          lastName: '',
+          address: '',
+          city: '',
+          country: 'US',
+          cardExpirationMonth: '',
+          cardExpirationYear: '',
+          optionalField: '',
+        },
+      ];
+
+      testFormValues.forEach((formValues) => {
+        const requiredFields = formValidators.getRequiredFields(formValues);
+        if (formValues.country) {
+          const { optionalField, ...expectedRequiredFields } = formValues;
+          expect(requiredFields).toEqual(expectedRequiredFields);
+        } else {
+          const { optionalField, ...expectedRequiredFields } = formValues;
+          expect(requiredFields).toEqual(expectedRequiredFields);
+        }
+      });
+    });
+
+    it('returns organization fields for a bulk order', () => {
+      const isBulkOrder = true;
+      mount((
+        <IntlProvider locale="en">
+          <AppContext.Provider value={{ authenticatedUser }}>
+            <reactRedux.Provider store={store}>
+              <Elements stripe={mockStripe}>
+                <StripePaymentForm
+                  isBulkOrder
+                  handleSubmit={() => {}}
+                  onSubmitPayment={() => {}}
+                  onSubmitButtonClick={() => {}}
+                />
+              </Elements>
+            </reactRedux.Provider>
+          </AppContext.Provider>
+        </IntlProvider>
+      ));
+
+      const formValues = {
+        firstName: '',
+        lastName: '',
+        address: '',
+        city: '',
+        country: 'UK',
+        cardNumber: '',
```

Comment:
Same comment as above, could we remove the credit card related fields since we don't get the values for them (goes directly to Stripe)?

Reply:
removed